state/policy.go

```diff
@@ -63,11 +63,7 @@ func (st *State) ResolveConstraints(cons constraints.Value) (constraints.Value,
 	if err != nil {
 		return constraints.Value{}, err
 	}
-	modelCons, err := st.ModelConstraints()
```

Comment:
ResolveConstraints is called in 4 places - when adding a machine and various places when adding an application or unit. Have we checked whether it's possible yet to make those call sites use the domain services.

Reply:
The application Jira card linked above is covering this. Have also made JUJU-7622 to track this from the machine's side of the equation.

Reply:
Noted with thanks tom and Ian :)